sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -391,10 +399,18 @@ object StateStore extends Logging {
     require(version >= 0)
     val storeProvider = loadedProviders.synchronized {
       startMaintenanceIfNeeded()
+
+      val newProvIdSchemaCheck = StateStoreProviderId.withNoPartitionInformation(storeProviderId)
+      if (!schemaValidated.contains(newProvIdSchemaCheck)) {
```

Comment:
Seems we should check `stateSchemaCheckEnabled` here. Then if it's disabled, we will not send extra RPC requests?

Reply:
So there're two different purposes on constructing schema information:

1) checking schema compatibility
2) leveraging schema information to open the chance for further improvements - e.g. read state without query (like state datasource), possibly applying projection to match schemas when if ordinal only differs, schema evolution (probably?)

We'd want to leave schema information even with the config being disabled, to enable 2).

Reply:
Let's hear more voice on the benefit of 2). If we don't reach consensus of benefits on 2), we can skip all the things if the config is disabled.

Reply:
+1 to fully disable this since the user explicitly asks. For example, they may hit some performance regression due to the extra RPC.

Reply:
It will be really odd if they're flipping the config during multiple runs and schema somehow changes in compatible way. It won't break compatibility check as the compatibility check is transitive (if I'm not mistaken), but once the compatibility is broken we may show legacy schema instead of the one for previous batch.

Probably I should just apply the trick to check this in reserved partition only (0) and don't trigger RPC.